### PR TITLE
added example of plotting positions on blank maps

### LINKED ISSUE #3241: Add an example on how to plot positions on a blank map
### Description

It is commonly seen in papers and presentations to show hpc positions of events such as flares on a blank (white) map. Because the events usually occurred at different times no single observation is appropriate.  

### Additional context

None

### PR DESCRIPTION
**Pull request by m11dedhia**
### Description

Added an example on how to plot positions on blank maps

Fixes #3241

### CODE REVIEW

File: .mailmap
Comment (nabobalis) [inline .mailmap:154]: I think you might want to undo this change.

File: examples/plotting/plotting_blank_map.py
Comment (nabobalis) [inline examples/plotting/plotting_blank_map.py:5]:
```suggestion
=============================
Plot positions on a blank map
=============================
```

File: .mailmap
Comment (m11dedhia) [reply .mailmap:154]: yeah, I'll do that, sorry for the inconvenience

File: examples/plotting/plotting_blank_map.py
Comment (nabobalis) [inline examples/plotting/plotting_blank_map.py:6]:
```suggestion

This example showcases how to plot positions on a blank map.
It is commonly seen in papers and presentations to show HPC positions of events that occurred at different times and therefore no single observation is appropriate.
```

File: examples/plotting/plotting_blank_map.py
Comment (nabobalis) [inline examples/plotting/plotting_blank_map.py:8]:
```suggestion
```

File: examples/plotting/plotting_blank_map.py
Comment (nabobalis) [inline examples/plotting/plotting_blank_map.py:21]:
```suggestion
# First we will create a blank map using with an array of zeros.
# Since there is no WCS information, we will need to construct a header to pass to Map.
```

File: examples/plotting/plotting_blank_map.py
Comment (nabobalis) [inline examples/plotting/plotting_blank_map.py:37]: Do we need to do this?

File: examples/plotting/plotting_blank_map.py
Comment (nabobalis) [inline examples/plotting/plotting_blank_map.py:33]:
```suggestion
# Now we have constructed the map, we can plot it and mark important locations to it. 
```

File: examples/plotting/plotting_blank_map.py
Comment (nabobalis) [inline examples/plotting/plotting_blank_map.py:54]: Why add a colorbar?

File: examples/plotting/plotting_blank_map.py
Comment (nabobalis) [inline examples/plotting/plotting_blank_map.py:51]: I think this title needs to be nicer.

File: examples/plotting/plotting_blank_map.py
Comment (m11dedhia) [reply examples/plotting/plotting_blank_map.py:37]: this defines the plot right?

File: examples/plotting/plotting_blank_map.py
Comment (m11dedhia) [reply examples/plotting/plotting_blank_map.py:54]: It's just plain black without a colorbar, would that be fine?

File: examples/plotting/plotting_blank_map.py
Comment (m11dedhia) [reply examples/plotting/plotting_blank_map.py:51]: what changes should I make?

File: examples/plotting/plotting_blank_map.py
Comment (nabobalis) [reply examples/plotting/plotting_blank_map.py:54]: Well its a blank map, why would it need a colorbar? 

Also if it's all 0s, it should made to be a white map? I am not sure if I prefer it to be a black or white map.

File: examples/plotting/plotting_blank_map.py
Comment (nabobalis) [reply examples/plotting/plotting_blank_map.py:51]: A better title?

File: examples/plotting/plotting_blank_map.py
Comment (m11dedhia) [reply examples/plotting/plotting_blank_map.py:54]: If it's a zero then it corresponds to a black. I'll remove the colorbar.

Comment (nabobalis) [reply examples/plotting/plotting_blank_map.py]: I assume we could rescale the map so it was white instead?

Comment (m11dedhia) [reply examples/plotting/plotting_blank_map.py]: I am not able to figure this out, could you please help me?

Comment (nabobalis) [reply examples/plotting/plotting_blank_map.py]: You mean changing the scale to white? 

I am not really sure how to do that, https://docs.sunpy.org/en/stable/generated/gallery/plotting/map_editcolormap.html#sphx-glr-generated-gallery-plotting-map-editcolormap-py might offer some insight to changing the norm or color map to acheive this. 

Otherwise we just leave it black.

Comment (hayesla) [reply examples/plotting/plotting_blank_map.py]: if you set alpha=0 in `plot()` it should be white.

File: examples/plotting/plotting_blank_map.py
Comment (hayesla) [inline examples/plotting/plotting_blank_map.py:27]:
```suggestion
header = sunpy.map.make_fitswcs_header(data, skycoord, scale=[2,2]*u.arcsec/u.pixel)
```

If you want the solar limb to be in the field of view (i.e. so you can use blank_map.draw_limb() and see it) then you need either the data array to be larger (the radius of Sun is ~960'') or you can set the pixel scale to be two arcsec per pixel as suggested here.

File: examples/plotting/plotting_blank_map.py
Comment (hayesla) [reply examples/plotting/plotting_blank_map.py:39]: you dont need to `im=` you can just do `blank.plot` as you are not using a colorbar

File: examples/plotting/plotting_blank_map.py
Comment (hayesla) [inline examples/plotting/plotting_blank_map.py:40]:
```suggestion
blank_map.draw_limb(color="k")
blank_map.draw_grid(color="k")
```
this will draw the solar limb (i.e. the disk of the Sun) and the grid.

**Review by hayesla — CHANGES_REQUESTED** [review]

**Comment by hayesla** [comment]
this is looking good - with some suggested changes it should now look like this : 
Screenshot 2021-03-12 at 17 34 31

File: examples/plotting/plotting_blank_map.py
Comment (nabobalis) [inline examples/plotting/plotting_blank_map.py:5]:
```suggestion

This example showcases how to plot positions on a blank map.
```

File: examples/plotting/plotting_blank_map.py
Comment (nabobalis) [inline examples/plotting/plotting_blank_map.py:6]:
```suggestion
It is commonly seen in papers and presentations to show HPC positions of events that occurred at
different times and therefore no single observation is appropriate.
```

**Comment by m11dedhia** [comment]
Can we merge this PR?

**Comment by nabobalis** [comment]
> Can we merge this PR?

We can when its ready. It still needs an approval and a changelog entry.

File: examples/plotting/plotting_blank_map.py
Comment (nabobalis) [inline examples/plotting/plotting_blank_map.py:35]:
```suggestion
```

File: examples/plotting/plotting_blank_map.py
Comment (nabobalis) [inline examples/plotting/plotting_blank_map.py:29]: I think there should be a comment about why we set the scale like this.

File: examples/plotting/plotting_blank_map.py
Comment (nabobalis) [inline examples/plotting/plotting_blank_map.py:31]:
```suggestion
# Use sunpy.map.Map to create the blank map
```

File: examples/plotting/plotting_blank_map.py
Comment (nabobalis) [inline examples/plotting/plotting_blank_map.py:57]:
```suggestion
ax.set_title('Plotting locations on a blank map')
```

**Comment by m11dedhia** [comment]
> > Can we merge this PR?
> 
> We can when its ready. It still needs an approval and a changelog entry.

This will be named as 5077.doc.rst in changelog right?

**Comment by nabobalis** [comment]
> > > Can we merge this PR?
> > 
> > 
> > We can when its ready. It still needs an approval and a changelog entry.
> 
> This will be named as 5077.doc.rst in changelog right?

Yeah that should work.

**Comment by m11dedhia** [comment]
How do I pass the three checks that are remaining?

**Comment by nabobalis** [comment]
The one you need to is https://dev.azure.com/sunpy/sunpy/_build/results?buildId=14871&view=logs&jobId=673f7616-04ea-52e2-247c-12c8f27bc080&j=673f7616-04ea-52e2-247c-12c8f27bc080&t=4088a903-e138-5bca-3a5d-42ba61a9adf0

Specifically that check is a linter:

```shell
diff --git a/changelog/5077.doc.rst b/changelog/5077.doc.rst
index 16ec98378..495e268ea 100644
--- a/changelog/5077.doc.rst
+++ b/changelog/5077.doc.rst
@@ -1 +1 @@
-Added an example `plotting_blank_map.py` on how to plot on a blank map
\ No newline at end of file
+Added an example `plotting_blank_map.py` on how to plot on a blank map
diff --git a/examples/plotting/plotting_blank_map.py b/examples/plotting/plotting_blank_map.py
index 987870334..37d901214 100644
--- a/examples/plotting/plotting_blank_map.py
+++ b/examples/plotting/plotting_blank_map.py
@@ -11,12 +11,11 @@ different times and therefore no single observation is appropriate.
 import matplotlib.pyplot as plt
 import numpy as np
 
-import sunpy.map
-from sunpy.coordinates import frames
-
 import astropy.units as u
 from astropy.coordinates import SkyCoord
-from astropy.io import fits
+
+import sunpy.map
+from sunpy.coordinates import frames
```

So the imports in the example need to be sorted and there needs to be a new line added to the changelog file.

Running `tox -e codestyle` should fix this locally and then you can commit it.

(We have some docs on this here: https://docs.sunpy.org/en/latest/dev_guide/contents/code_standards.html#formatting).

File: CITATION.rst
Comment (nabobalis) [inline CITATION.rst:1]: Why was this changed?

Comment (m11dedhia) [reply CITATION.rst:1]: the pre commit made these changes automatically, should I reverse it back?

Comment (nabobalis) [reply CITATION.rst:1]: That is strange, it doesn't need to one master. Leave it in for now.

Comment (m11dedhia) [reply CITATION.rst:1]: okay now what should I do now?

Comment (nabobalis) [reply CITATION.rst:1]: Nothing, wait for reviews and then once this PR has been approved we will merge it.

File: examples/plotting/plotting_blank_map.py
Comment (nabobalis) [inline examples/plotting/plotting_blank_map.py:56]:
```suggestion
ax.set_title('Plotting fixed points on a blank map')
```

File: changelog/5077.doc.rst
Comment (nabobalis) [inline changelog/5077.doc.rst:1]:
```suggestion
Added a gallery example  (:ref:`sphx_glr_generated_gallery_plotting_plotting_blank_map.py`) on how to create a blank map and mark locations.
```

**Comment by m11dedhia** [comment]
> The one you need to fix (aka `sunpy.sunpy (Core Tests style_check [linux])`) is https://dev.azure.com/sunpy/sunpy/_build/results?buildId=14871&view=logs&jobId=673f7616-04ea-52e2-247c-12c8f27bc080&j=673f7616-04ea-52e2-247c-12c8f27bc080&t=4088a903-e138-5bca-3a5d-42ba61a9adf0
> 
> Specifically that check is a linter:
> 
> ```shell
> diff --git a/changelog/5077.doc.rst b/changelog/5077.doc.rst
> index 16ec98378..495e268ea 100644
> --- a/changelog/5077.doc.rst
> +++ b/changelog/5077.doc.rst
> @@ -1 +1 @@
> -Added an example `plotting_blank_map.py` on how to plot on a blank map
> \ No newline at end of file
> +Added an example `plotting_blank_map.py` on how to plot on a blank map
> diff --git a/examples/plotting/plotting_blank_map.py b/examples/plotting/plotting_blank_map.py
> index 987870334..37d901214 100644
> --- a/examples/plotting/plotting_blank_map.py
> +++ b/examples/plotting/plotting_blank_map.py
> @@ -11,12 +11,11 @@ different times and therefore no single observation is appropriate.
>  import matplotlib.pyplot as plt
>  import numpy as np
>  
> -import sunpy.map
> -from sunpy.coordinates import frames
> -
>  import astropy.units as u
>  from astropy.coordinates import SkyCoord
> -from astropy.io import fits
> +
> +import sunpy.map
> +from sunpy.coordinates import frames
> ```
> 
> So the imports in the example need to be sorted and there needs to be a new line added to the changelog file.
> 
> Running `tox -e codestyle` should fix this locally and then you can commit it.
> 
> (We have some docs on this here: https://docs.sunpy.org/en/latest/dev_guide/contents/code_standards.html#formatting).

The Tox command you mentioned is not working

**Comment by nabobalis** [comment]
What is the issue with tox? 

There is nothing to correct so it shouldn't do anything.

**Comment by m11dedhia** [comment]
2 checks still failed what do i do about them?

**Comment by nabobalis** [comment]
I think you might need to undo the change to "CITATION.rst" to fix it.

File: CITATION.rst
Comment (nabobalis) [inline CITATION.rst:1]: Ideally this file should not show up on the files changed tab of github. You will want to check out this file from the main sunpy repository.

**Comment by nabobalis** [comment]
If you can, I would also suggest rebasing this on top of the current sunpy master.

**Comment by m11dedhia** [comment]
should i do a git pull request?

**Comment by nabobalis** [comment]
> should i do a git pull request?

I am not sure I follow, you could do a git pull instead of a rebase.

**Comment by m11dedhia** [comment]
No I thought i could rebase using git pull, how should i rebase then?

**Comment by nabobalis** [comment]
> No I thought i could rebase using git pull, how should i rebase then?

I mean if you can, why not.

**Comment by m11dedhia** [comment]
what other options do i have? Like what should CITATION.rst contain?

**Comment by nabobalis** [comment]
> what other options do i have? Like what should CITATION.rst contain?

It should contain the same thing as on sunpy master. Ideally it would not be changed in your pull request.

**Comment by nabobalis** [comment]
Current rendered version: https://sunpy--5077.org.readthedocs.build/en/5077/generated/gallery/plotting/plotting_blank_map.html#sphx-glr-generated-gallery-plotting-plotting-blank-map-py

**Review by nabobalis — APPROVED** [review]

File: examples/plotting/plotting_blank_map.py
Comment (ayshih) [inline examples/plotting/plotting_blank_map.py:23]: Not a big deal, but why use `np.full()` instead of `np.zeros()`?

File: examples/plotting/plotting_blank_map.py
Comment (ayshih) [inline examples/plotting/plotting_blank_map.py:40]: The need for `alpha=0` should be explained in a comment

File: examples/plotting/plotting_blank_map.py
Comment (ayshih) [inline examples/plotting/plotting_blank_map.py:45]: Is this vestigial?  The coordinate points you are plotting shouldn't trigger any re-scaling.

File: examples/plotting/plotting_blank_map.py
Comment (m11dedhia) [reply examples/plotting/plotting_blank_map.py:23]: Okay, will change that

File: examples/plotting/plotting_blank_map.py
Comment (m11dedhia) [reply examples/plotting/plotting_blank_map.py:45]: okay, I'll remove it

File: examples/plotting/plotting_blank_map.py
Comment (m11dedhia) [reply examples/plotting/plotting_blank_map.py:40]: Added 'Plot the map with alpha=0 to provide white plot' in the comments

**Comment by m11dedhia** [comment]
How do i see what's going wrong in the codecov test?

**Comment by nabobalis** [comment]
Codecov failed due to the core tests failing. It isn't something to worry about.

**Comment by m11dedhia** [comment]
I installed tox locally and still it doesn't recognize tox as an internal or external command. I refered to this site:
https://tox.readthedocs.io/en/latest/index.html

**Comment by nabobalis** [comment]
> I installed tox locally and still it doesn't recognize tox as an internal or external command. I refered to this site:
> [tox.readthedocs.io/en/latest/index.html](https://tox.readthedocs.io/en/latest/index.html)

What was the error? Did you create a virtual environment and install tox inside of that?

**Comment by m11dedhia** [comment]
I don't know how to create a virtual environment, could you refer some docs from where i can read?

**Comment by nabobalis** [comment]
> I don't know how to create a virtual environment, could you refer some docs from where i can read?

https://docs.sunpy.org/en/latest/dev_guide/contents/newcomers.html should cover all of that.

File: examples/plotting/plotting_blank_map.py
Comment (Cadair) [reply examples/plotting/plotting_blank_map.py:42]: If the array is all `NaN`s then I don't think you need this.

File: examples/plotting/plotting_blank_map.py
Comment (Cadair) [inline examples/plotting/plotting_blank_map.py:23]: This is a reasonably large array, when it really doesn't need to be. A (10, 10) array would suffice, as you just scale the pixel size.

Also as I mentioned below I think this should be an array of `NaN`:
```suggestion
data = np.zeros((1000, 1000)) * np.nan
```

Comment (ayshih) [reply examples/plotting/plotting_blank_map.py:23]: Ah, an array full of `NaN` makes me happier.  In that case, definitely go back to using `np.full()`:
```suggestion
data = np.full((1000, 1000), np.nan)
```

**Comment by m11dedhia** [comment]
tox runs perfectly in my virtual environment, and all codestlye commands succeeded too, what should i do now?

**Comment by nabobalis** [comment]
Nothing.

**Comment by m11dedhia** [comment]
So we can merge the pull request?

**Comment by nabobalis** [comment]
> So we can merge the pull request?

No, you will want to address the outstanding comments e.g., using a smaller array and increasing the scale, removing the need for alpha if it is white with an all nan array

**Comment by m11dedhia** [comment]
Smaller array as in just one point that should be plotted?

**Comment by nabobalis** [comment]
Yeah, it doesn't need to be 1000 by 1000.

File: CITATION.rst
Comment (nabobalis) [inline CITATION.rst:1]: I still think there should be no difference in this file. It is a symlink in the repo and I am weary of a change here.

File: examples/plotting/plotting_blank_map.py
Comment (nabobalis) [inline examples/plotting/plotting_blank_map.py:8]:
```suggestion
It is commonly used to show HelioProjective Cartesian (HPC) positions of events that occurred at
different times and therefore no single observation is appropriate.
```

File: examples/plotting/plotting_blank_map.py
Comment (Cadair) [inline examples/plotting/plotting_blank_map.py:23]:
```suggestion
data = np.full((10, 10), np.nan)
```

File: examples/plotting/plotting_blank_map.py
Comment (Cadair) [inline examples/plotting/plotting_blank_map.py:29]:
```suggestion
header = sunpy.map.make_fitswcs_header(data, skycoord, scale=[220, 220]*u.arcsec/u.pixel)
```

**Review by Cadair — COMMENTED** [review]
Shrinking the array needs to come with a increase in scale, the sun is a smidge over 2000" in diameter, so the scale * shape should be over 2000".

File: CITATION.rst
Comment (m11dedhia) [reply CITATION.rst:1]: Ya I know, i just added whats there in the master branch in sunpy

Comment (nabobalis) [reply CITATION.rst:1]: There should be no changes to this file in your pull request. It should be the same as the code in master.

Comment (m11dedhia) [reply CITATION.rst:1]: I rebased the whole thing, created a new branch with resetting commits prior to editing this file and i have merged into m11dedhia/sunpy master repository. Will this do?

Comment (nabobalis) [reply CITATION.rst:1]: No since the file has still changed in this PR.

Comment (nabobalis) [reply CITATION.rst:1]: Closing and opening a new PR won't change anything.

If you follow: https://stackoverflow.com/questions/1817766/how-to-revert-to-origins-master-branchs-version-of-file you should be able to reset this file.

**Comment by m11dedhia** [comment]
I think this will work, did the needful

File: examples/plotting/plotting_blank_map.py
Comment (hayesla) [inline examples/plotting/plotting_blank_map.py:8]: I think this sentence is a bit confusing, and this example could also be extended to plot other points that are not in HPC coordinate system (i.e. just define a different SkyCoord). How about:

```suggestion
It is often useful to plot coordinate positions of events on a blank helioprojective coordinate map. 
In this example, we create an empty map with a WCS defined by a helioprojective frame as observed from Earth at a certain time, and show how you can plot different coordinates on it.
```

File: examples/plotting/plotting_blank_map.py
Comment (hayesla) [inline examples/plotting/plotting_blank_map.py:25]:
```suggestion
# Define a reference coordinate and create a header using sunpy.map.make_fitswcs_header
```

**Review by hayesla — APPROVED** [review]

**Comment by hayesla** [comment]
looks great - just some minor suggested edits @m11dedhia

**Comment by nabobalis** [comment]
Thanks again @m11dedhia

**Comment by nabobalis** [comment]
This has been manually backported.

**Pull request merged**
